src/compiler/transclude.js

```diff
@@ -87,11 +87,13 @@ function transcludeContent (el, raw) {
       select = outlet.getAttribute('select')
       if (select) {  // select content
         selected = raw.querySelectorAll(select)
-        outlet.content = _.toArray(
-          selected.length
-            ? selected
-            : outlet.childNodes
-        )
+        if (selected.length) {
```

Comment:
I wanted to avoid element.matches because its browser support is slightly worse than querySelectorAll (http://caniuse.com/#feat=matchesselector vs http://caniuse.com/#search=queryselectorall). In practice it probably doesn't matter (opera mini 8 is the only browser that's still used of those, and I don't know if vue even supports it), but I decided to play it safe.